compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/viewinterop/UIKitInteropListener.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.viewinterop
+
+import kotlinx.cinterop.CValue
+import platform.CoreGraphics.CGSize
+
+/**
+ * An interface containing methods related to events emitted when an interop view transits to
+ * specific states. All the changes are executed synchronously right after the state change on a
+ * main thread in sync with a CATransaction batching UIKit objects changes to sync it with Compose
+ * rendering.
+ */
+interface UIKitInteropListener<T> {
```

Comment:
Is this the final shape of this API? public-stable is intended here?

Reply:
Actually, I've requested some insight from our dev advocates, so it could be now